### DISCUSSION TITLE
QueryDSL 빌드 실패

### DISCUSSION THREAD

**Discussion by umuno** [100-hours-a-week/discussions#7] [개발] [answered]
![image](https://github.com/user-attachments/assets/d7d1df85-7a63-430c-a6c8-d5dab55d18e5)

### 원인
1. 문제의 원인은 build.gradle 파일에서 QueryDSL 설정에 있었습니다.
2. 특히 implementation과 annotationProcessor로 선언된 QueryDSL 의존성을 점검한 결과, id "com.ewerk.gradle.plugins.querydsl" version "1.0.10" 플러그인 설정으로 인해 빌드 옵션에서 루트 경로를 지정하는 설정이 필요했습니다.
3. 이 설정이 없어서 빌드 에러가 발생한 것입니다.

![image](https://github.com/user-attachments/assets/193d5e8f-681d-43e4-b5dd-8db3e6db8af4)

### 결론
* build.gradle 파일에서 QueryDSL 빌드 옵션의 루트 지정 부분을 주석 처리하여, 빌드가 자동으로 진행되도록 수정했습니다.
* 그 결과, 빌드 에러를 해결할 수 있었습니다.
![image](https://github.com/user-attachments/assets/842fce2b-4095-4719-ad23-11efc7d13ce5)

**Comment by ej31** [comment]
- 마크다운의 문법 중 백틱 (`) 으로 단어를 감싸면 코드처럼 보이게 하는 기능이 있습니다.
- 그리고 하이픈 (-) 혹은 숫자("1. " 과 같이)로 문장을 시작하면 개조식으로 끊어서 설명이 가능합니다.
- 이러한 문법을 활용해서 조금 더 가독성 있게 작성해주시면 좋겠습니다. 🤡

**Comment by ej31** [comment, marked as answer]
답변 내용 정리 잘 하셨습니다~ 굿굿

**Comment by ej31** [comment]
### 원인
1. 문제의 원인은 `build.gradle` 파일에서 QueryDSL 설정에 있었습니다.
2. 특히 `implementation`과 `annotationProcessor`로 선언된 QueryDSL 의존성을 점검한 결과, `id "com.ewerk.gradle.plugins.querydsl" version "1.0.10"` 플러그인 설정으로 인해 빌드 옵션에서 루트 경로를 지정하는 설정이 필요했습니다.
3. 이 설정이 없어서 빌드 에러가 발생한 것입니다.

### 결론
- `build.gradle` 파일에서 QueryDSL 빌드 옵션의 루트 지정 부분을 주석 처리하여, 빌드가 자동으로 진행되도록 수정했습니다.
- 그 결과, 빌드 에러를 해결할 수 있었습니다.
